crates/astria-sequencer-types/src/sequencer_block_data.rs

```diff
@@ -328,7 +328,8 @@ impl SequencerBlockData {
         }
 
         // generate the action tree root proof of inclusion in `Header::data_hash`
-        let tx_tree = MerkleTree::from_leaves(b.data);
+        let hashed_txs = b.data.iter().map(|tx| sha256_hash(tx)).collect::<Vec<_>>();
```

Comment:
In a follow-up the merkle tree should be generalized to take `[u8; 32]` to avoid all the (now unnecessary) allocations.